verkle-trie/src/database/meta.rs

```diff
@@ -11,86 +12,89 @@ pub struct StemMeta {
     pub hash_stem_commitment: Fr,
 }
-
 impl std::fmt::Debug for StemMeta {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("StemMeta")
-            .field("c_1", &hex::encode(compress_point_to_array(&self.c_1)))
-            .field("c_2", &hex::encode(compress_point_to_array(&self.c_2)))
-            .field("hash_c1", &hex::encode(scalar_to_array(&self.hash_c1)))
-            .field("hash_c2", &hex::encode(scalar_to_array(&self.hash_c2)))
+            .field("c_1", &hex::encode(compress_point_to_array(&self.c_1).unwrap()))
+            .field("c_2", &hex::encode(compress_point_to_array(&self.c_2).unwrap()))
+            .field("hash_c1", &hex::encode(scalar_to_array(&self.hash_c1).unwrap()))
+            .field("hash_c2", &hex::encode(scalar_to_array(&self.hash_c2).unwrap()))
             .field(
                 "stem commitment",
-                &hex::encode(compress_point_to_array(&self.stem_commitment)),
+                &hex::encode(compress_point_to_array(&self.stem_commitment).unwrap()),
             )
             .field(
                 "hash_stem_commitment",
-                &hex::encode(scalar_to_array(&self.hash_stem_commitment)),
+                &hex::encode(scalar_to_array(&self.hash_stem_commitment).unwrap()),
             )
             .finish()
     }
 }
 
-fn point_to_array(p: &Element) -> [u8; 64] {
+use ark_serialize::CanonicalSerialize;
+
+fn point_to_array(p: &Element) -> Result<[u8; 64], SerializationError> {
     let mut bytes = [0u8; 64];
-    use ark_serialize::CanonicalSerialize;
-    p.serialize_uncompressed(&mut bytes[..]).unwrap();
-    bytes
+    p.serialize_uncompressed(&mut bytes[..])?;
+
+    Ok(bytes)
 }
-fn compress_point_to_array(p: &Element) -> [u8; 32] {
+fn compress_point_to_array(p: &Element) -> Result<[u8; 32], SerializationError> {
     let mut bytes = [0u8; 32];
-    use ark_serialize::CanonicalSerialize;
-    p.serialize(&mut bytes[..]).unwrap();
-    bytes
+    p.serialize(&mut bytes[..])?;
+    
+    Ok(bytes)
 }
-fn scalar_to_array(scalar: &Fr) -> [u8; 32] {
+fn scalar_to_array(scalar: &Fr) -> Result<[u8; 32], SerializationError> {
     let mut bytes = [0u8; 32];
-    use ark_serialize::CanonicalSerialize;
-    scalar.serialize_uncompressed(&mut bytes[..]).unwrap();
-    bytes
+    scalar.serialize_uncompressed(&mut bytes[..])?;
+
+    Ok(bytes)
 }
-impl StemMeta {
+
+impl FromBytes<[u8; 288]> for StemMeta {
     // panic if we cannot deserialise, do not call this method if you are unsure if the data is
-    // not structured properly. We cn guarantee this in verkle trie.
-    pub fn from_bytes(bytes: &[u8]) -> StemMeta {
+    // not structured properly. We can guarantee this in verkle trie.
+    fn from_bytes(bytes: [u8; 288]) -> Result<StemMeta, SerializationError> {
         let len = bytes.len();
-        assert_eq!(len, 64 * 3 + 32 * 3);
-
-        use ark_serialize::CanonicalDeserialize;
+        if !(len == 64 * 3 + 32 * 3) {
```

Comment:
This if statement seems redundant, if bytes is an array with its type defined to be 288.

Since we are putting the length outside of the function, a minor note for another PR is to explain where the 288 comes from so it doesn't look like a magic number

Reply:
Had to convert it back to a `Vec` for current type constraints, will add a `TODO` though

Reply:
Not sure I understand where vector comes into play here, the length of that array is always 288.

Reply:
It's an ugly artifact of on-going trait refactor, I tried to make a generic `ToBytes` trait which used Generic slices but didn't work in every instance so for now Generic Slices got moved to a `Vec` until that refactoring can be more cleanly added